src/plugin_manager/basic_plugin_wrapper.cpp

```diff
@@ -0,0 +1,181 @@
+/*
+ * Punica - LwM2M server with REST API
+ * Copyright (C) 2019 8devices
+ *
+ * This program is free software: you can redistribute it and/or modify it
+ * under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License,
+ * or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ * See the GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program. If not, see <https://www.gnu.org/licenses/>.
+ *
+ */
+
+#include <cstring>
+#include <dlfcn.h>
+#include <string.h>
+
+#include <punica/version.h>
+
+#include "basic_plugin_wrapper.hpp"
+
+#include "../logging.h"
+
+static const char loggingSection[] = "[PLUGINS]";
+
+BasicPluginWrapper::BasicPluginWrapper(std::string path,
+                                       std::string name):
+    mDLL(nullptr),
```

Comment:
question: how should we deal with abbreviations in camel case? HTTPResponse or HttpResponse?